unit_testing/test_labels.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# pytest unit tests for spinalcordtoolbox.labels
+
+import logging
+from time import time
+
+import numpy as np
+import pytest
+
+import spinalcordtoolbox.labels as sct_labels
+from spinalcordtoolbox.image import Image, zeros_like
+from spinalcordtoolbox.utils import sct_test_path
+from spinalcordtoolbox.types import Coordinate
+from test_image import fake_3dimage, fake_3dimage2
+
+logger = logging.getLogger(__name__)
+
+seg_img = Image(sct_test_path('t2', 't2_seg-manual.nii.gz'))
+t2_img = Image(sct_test_path('t2', 't2.nii.gz'))
+labels_img = Image(sct_test_path('t2', 'labels.nii.gz'))
+
+
+# TODO [AJ] investigate how to parametrize fixtures from test_image.py
+# without redefining the function here
+def fake_3dimage_sct2():
+    """
+    :return: an Image (3D) in RAS+ (aka SCT LPI) space
+    shape = (1,2,3)
+    """
+    i = fake_3dimage2()
+    img = Image(i.get_data(), hdr=i.header,
+                orientation="RPI",
+                dim=i.header.get_data_shape(),
+                )
+    return img
+
+
+def fake_3dimage_sct():
+    """
+    :return: an Image (3D) in RAS+ (aka SCT LPI) space
+    shape = (7,8,9)
+    """
+    i = fake_3dimage()
+    img = Image(i.get_data(), hdr=i.header,
+                orientation="LPI",
+                dim=i.header.get_data_shape(),
+                )
+    return img
+
+
+test_images = [fake_3dimage_sct(), fake_3dimage_sct2(), t2_img]
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_create_labels_empty(test_image):
+    a = test_image.copy()
+    expected = zeros_like(a)
+
+    labels = [Coordinate(l) for l in [[0, 0, 0, 7], [0, 1, 2, 5]]]
+    expected.data[0, 0, 0] = 7
+    expected.data[0, 1, 2] = 5
+
+    b = sct_labels.create_labels_empty(a, labels)
+
+    diff = b.data == expected.data
+    assert diff.all()
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_create_labels(test_image):
+    a = test_image.copy()
+    labels = [Coordinate(l) for l in [[0, 1, 0, 99], [0, 1, 2, 5]]]
+
+    b = sct_labels.create_labels(a, labels)
+
+    assert b.data[0, 1, 0] == 99
+    assert b.data[0, 1, 2] == 5
+
+
+@pytest.mark.parametrize("test_seg", [seg_img])
+def test_create_labels_along_segmentation(test_seg):
+    a = test_seg.copy()
+    labels = [(5, 1), (14, 2), (23, 3)]
+
+    b = sct_labels.create_labels_along_segmentation(a, labels)
+
+    assert b.orientation == a.orientation
+    # TODO [AJ] implement test
```

Comment:
simple way (assumes test data don't change):
- run the test and see where the output labels are, then hard-code the labels coordinates and verify that they match the output of this test (e.g. image difference) 

complex way
- compute center of mass at the indicated slices, and verify that those correspond to the output labels